### PR TITLE
Unit tests for GetSimpleEventFilter

### PR DESCRIPTION
**Pull request by peterbryntesson**

### CODE REVIEW

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/tests/Services/EventTestsBase.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+﻿namespace Microsoft.Azure.IIoT.OpcUa.Protocol.Services {
```
Comment (simonjaeger) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/tests/Services/EventTestsBase.cs:1]: Missing copyright header.

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/tests/Services/EventTestsBase.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    using Microsoft.Azure.IIoT.OpcUa.Protocol.Models;
+    using Moq;
+    using Opc.Ua;
+    using Opc.Ua.Client;
+    using Serilog;
+    using static Microsoft.Azure.IIoT.OpcUa.Protocol.Services.SubscriptionServices;
+
+    public class EventTestsBase {
```
Comment (simonjaeger) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/tests/Services/EventTestsBase.cs:9]: Could be marked abstract.

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/tests/Services/GetSimpleEventFilterTests.cs [hunk 3]
```diff
@@ -10,13 +10,11 @@ namespace Microsoft.Azure.IIoT.OpcUa.Protocol.Tests.Services {
     using Moq;
```
Comment (simonjaeger) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/tests/Services/GetSimpleEventFilterTests.cs:11]: This class is in a different namespace than the other tests. Let's keep using `Microsoft.Azure.IIoT.OpcUa.Protocol.Services` for consistency.

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs [hunk 4]
```diff
@@ -1358,7 +1352,13 @@ public class MonitoredItemWrapper {
                 return Item.MonitoringMode == change ? null : change;
             }
 
-            internal EventFilter GetSimpleEventFilter(INodeCache nodeCache, ServiceMessageContext context) {
+            /// <summary>
+            /// Builds select clause and where clause by using opc ua reflection
```
Comment (simonjaeger) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs:1356]: Detail: Let's capitalize OPC UA.

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/tests/Services/GetSimpleEventFilterTests.cs [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+        public void SetupSimpleFilterForBaseEventType() {
+            // Arrange
+            var template = new EventMonitoredItemModel {
+                EventFilter = new EventFilterModel() {
+                    TypeDefinitionId = ObjectTypeIds.BaseEventType.ToString()
+                }
+            };
+
+            // Act
+            var monitoredItemWrapper = GetMonitoredItemWrapper(template, null, null);
```
Comment (simonjaeger) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/tests/Services/GetSimpleEventFilterTests.cs:29]: Redundant nulls.

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/tests/Services/GetSimpleEventFilterTests.cs [hunk 6]
```diff
@@ -0,0 +1,288 @@
...
+            Assert.NotNull(eventFilter.WhereClause.Elements[0].FilterOperands);
+            Assert.Single(eventFilter.WhereClause.Elements[0].FilterOperands);
+            Assert.IsType<LiteralOperand>(eventFilter.WhereClause.Elements[0].FilterOperands[0].Body);
+            var operand = (LiteralOperand)eventFilter.WhereClause.Elements[0].FilterOperands[0].Body;
+            Assert.IsType<NodeId>(operand.Value.Value);
+            var nodeId = (NodeId)operand.Value.Value;
+            Assert.Equal(nodeId, ObjectTypeIds.ConditionType);
+        }
+
+
```
Comment (simonjaeger) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/tests/Services/GetSimpleEventFilterTests.cs:113]: A redundant empty line.

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/tests/Services/GetSimpleEventFilterTests.cs [hunk 7]
```diff
@@ -0,0 +1,288 @@
...
+            _conditionTypeNode.ReferenceTable.Add(ReferenceTypeIds.HasComponent, false, _enabledStateNode.NodeId);
+            _enabledStateNode.ReferenceTable.Add(ReferenceTypeIds.HasComponent, true, ObjectTypeIds.ConditionType);
+            _enabledStateNode.ReferenceTable.Add(ReferenceTypeIds.HasProperty, false, _idNode.NodeId);
+            _idNode.ReferenceTable.Add(ReferenceTypeIds.HasProperty, true, _enabledStateNode.NodeId);
+            _commentNode.ReferenceTable.Add(ReferenceTypeIds.HasProperty, true, ObjectTypeIds.ConditionType);
+            nodeCache.Setup<Node>(x => x.FetchNode(It.IsAny<ExpandedNodeId>())).Returns((ExpandedNodeId x) => {
+                if (x.IdType == IdType.Numeric && x.Identifier is uint id) {
+                    return _nodes[id];
+                }
+                else {
```
Comment (simonjaeger) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/tests/Services/GetSimpleEventFilterTests.cs:205]: Detail: You don't need the else statement here.

**Review by simonjaeger — CHANGES_REQUESTED** [review]
Great stuff! Minor comments.

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/tests/Services/EventTestsBase.cs (hunk 1 above)
Comment (peterbryntesson) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/tests/Services/EventTestsBase.cs]: Fixed

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/tests/Services/EventTestsBase.cs (hunk 2 above)
Comment (peterbryntesson) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/tests/Services/EventTestsBase.cs:14]: Fixed

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/tests/Services/GetSimpleEventFilterTests.cs (hunk 3 above)
Comment (peterbryntesson) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/tests/Services/GetSimpleEventFilterTests.cs:10]: Fixed

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs (hunk 4 above)
Comment (peterbryntesson) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs:1356]: Fixed

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/tests/Services/GetSimpleEventFilterTests.cs (hunk 5 above)
Comment (peterbryntesson) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/tests/Services/GetSimpleEventFilterTests.cs:28]: Fixed

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/tests/Services/GetSimpleEventFilterTests.cs (hunk 6 above)
Comment (peterbryntesson) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/tests/Services/GetSimpleEventFilterTests.cs:112]: Fixed

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/tests/Services/GetSimpleEventFilterTests.cs (hunk 7 above)
Comment (peterbryntesson) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/tests/Services/GetSimpleEventFilterTests.cs:205]: Fixed

**Review by simonjaeger — APPROVED** [review]
LGTM

**Review by koepalex — APPROVED** [review]
:shipit:

**Pull request merged**